### ISSUE TITLE
MEMORY_LIMIT_EXCEEDED

### ISSUE THREAD

**Issue by LYJ1999904** [ClickHouse/ClickHouse#56356]
We use the ClickHouse container to allow our development team members to debug our application on their local machines. After upgrading to clickhouse/clickhouse-server:22.3.8.39, we encountered an issue where the container cannot start. The error message we receive is:

DB::MergeTreeBackgroundExecutor<DB::MergeMutateRuntimeQueue>::routine(DB::TaskRuntimeDataPtr) [Queue = DB::MergeMutateRuntimeQueue]: Code: 241. DB::Exception: Memory limit (total) exceeded: would use 493.54 MiB (attempt to allocate chunk of 4194306 bytes), maximum: 460.80 MiB. (MEMORY_LIMIT_EXCEEDED)

We attempted to increase the <max_memory_usage> and <max_server_memory_usage> values in both the docker/clickhouse/config.d/config.xml and docker/clickhouse/config.d/users.xml files to 2GB, but the issue still persists. Interestingly, this problem only occurs on Windows systems within our team, while Mac systems work fine."

**Comment by nickitat** [comment]
pls connect to the server and verify that changes were applied

``` sql
show settings like '%max%mem%'
```

**Comment by den-crane** [comment]
>maximum: 460.80 MiB.

>ls -lh /usr/bin/clickhouse
>-r-xr-xr-x 1 root root 505M Nov  3 16:30 /usr/bin/clickhouse

try 

```
/etc/clickhouse-server/config.d/mlock_executable.xml
<clickhouse>
      <mlock_executable>false</mlock_executable>
</clickhouse>
```

**Comment by LYJ1999904** [comment]
I just wanted to drop a quick note to say thanks for the help in getting our ClickHouse container up and running with a 2GB memory limit. We hit a few bumps along the way, and the community was awesome in helping us figure it out. Here's what we did:

In our deployment config, we set the memory limits to 2GB like this:

yaml
Copy code
deploy:
  resources:
    limits:
      memory: 2048M
    reservations:
      memory: 2048M
This setup allowed us to start the ClickHouse container with enough memory to handle our workload without any memory-related hiccups.

With the 2GB memory limit, our ClickHouse instance could smoothly handle our development and testing tasks. It's a solid choice for most dev and test scenarios.

Huge thanks to the community for the support and guidance in sorting out the memory limit issues. Your help was greatly appreciated, and it made a big difference in getting our dev environment up and running smoothly.
